### PR TITLE
Implement colorize

### PR DESCRIPTION
**Pull request by Yexeed**
## Introduction

this will make life easier when you dont should use TextFormat::<symbol> for every color-symbol to colorize the string
p.s. big thanks to SOFe, i dont know why he didnt make this PR earlier
### API changes

new static method in TextFormat
## Tests

```
public function onEnable(){
$this->getLogger()->info(TextFormat::colorize("&bExample &etext");
//or
$this->getLogger()->info(TextFormat::colorize("%bExample %etext", '%');
}
```

### CODE REVIEW

File: src/pocketmine/utils/TextFormat.php [hunk 1]
```diff
@@ -64,6 +64,17 @@
 	public static function tokenize(string $string) : array{
 		return preg_split("/(" . TextFormat::ESCAPE . "[0123456789abcdefklmnor])/", $string, -1, PREG_SPLIT_NO_EMPTY | PREG_SPLIT_DELIM_CAPTURE);
 	}
+	
+	/**
```
Comment (JackNoordhuis) [inline src/pocketmine/utils/TextFormat.php]: formatting

File: src/pocketmine/utils/TextFormat.php [hunk 2]
```diff
@@ -64,6 +64,17 @@
...
+     * Colorizes the string by replacing specified symbol to '§' ('&' by default)
+     *
+     * @param string $string
+     *
+     * @return string
+     */
+    public static function colorize(string $string, string $replaceSymb = '&'): string{
+        return preg_replace('/' . $replaceSymb . '([0-9a-fk-or])/u', TextFormat::ESCAPE . '$1', $string);
```
Comment (JackNoordhuis) [inline src/pocketmine/utils/TextFormat.php]: The rest of the code uses double quotes

File: src/pocketmine/utils/TextFormat.php (hunk 1 above)
Comment (Yexeed) [reply src/pocketmine/utils/TextFormat.php]: formatting, yeah. Doesnt get what do you mean :\

File: src/pocketmine/utils/TextFormat.php [hunk 3]
```diff
@@ -66,14 +66,14 @@ public static function tokenize(string $string) : array{
...
+     * Colorizes the string by replacing specified color symbol to "§" ("&" by default)
      *
      * @param string $string
      *
      * @return string
      */
-    public static function colorize(string $string, string $replaceSymb = '&'): string{
-        return preg_replace('/' . $replaceSymb . '([0-9a-fk-or])/u', TextFormat::ESCAPE . '$1', $string);
+    public static function colorize(string $string, string $replaceSymb = "&"): string{
+        return preg_replace("/" . $replaceSymb . "([0-9a-fk-or])/u", TextFormat::ESCAPE . "$1", $string);
```
Comment (SOF3) [inline src/pocketmine/utils/TextFormat.php]: Consider using preg_quote().

File: src/pocketmine/utils/TextFormat.php (hunk 1 above)
Comment (JackNoordhuis) [reply src/pocketmine/utils/TextFormat.php]: Wrong indentation, looks like an extra tab.

Comment (Yexeed) [reply src/pocketmine/utils/TextFormat.php]: how to fix?

File: src/pocketmine/utils/TextFormat.php (hunk 3 above)
Comment (Yexeed) [reply src/pocketmine/utils/TextFormat.php]: example, pls?

Comment (Yexeed) [reply src/pocketmine/utils/TextFormat.php]: preg_quote($replaceSymb) ??

File: src/pocketmine/utils/TextFormat.php (hunk 1 above)
Comment (Muqsit) [reply src/pocketmine/utils/TextFormat.php]: Remove this
screenshot_20171220-162009

**Comment by SOF3** [comment]
Let's close this since it is getting too difficult.

**Pull request closed without merging**